services/keybase/keybase-btc.service.js

```diff
@@ -60,7 +60,7 @@ module.exports = class KeybaseBTC extends KeybaseProfile {
 
   async handle({ username }) {
     const options = {
-      form: {
+      qs: {
```

Comment:
if we're changing these, could we go ahead and convert to got's `searchParams`

Reply:
Because it would throw `Found unrecognised options searchParams` :)

I think the reasons for that are explained in the top post and https://github.com/badges/shields/pull/6160#discussion_r575695268 but just to run over it again: My proposal for the migration plan is we keep everything communicating using request's `options` format regardless of whether we are actually using request or got (at least in the service code. Maybe the more 'standalone' stuff makes sense to bypass that). Then once we've removed request completely, the last stage of the migration will be:

- remove `requestOptions2GotOptions()`
- update everything to use got's syntax
- update https://github.com/badges/shields/blob/master/doc/TUTORIAL.md to point to got's docs instead of request's

I think allowing both a got-like options object or a request-like options object during the transition period probably introduces unnecessary complexity, but maybe it becomes useful to be able to break that down into smaller chunks nearer the time. I think the way I'd probably do that would be to do one parameter at a time over all the services rather than have some services using got's format and some using request's.

Reply:
> Because it would throw `Found unrecognised options searchParams` :)

Ah so it would. Apologies, have been reviewing this in small chunks over a long period of time and clearly forgetting things along the way.

> I think allowing both a got-like options object or a request-like options object during the transition period probably introduces unnecessary complexity, but maybe it becomes useful to be able to break that down into smaller chunks nearer the time

As I'm thinking about this (again :smile:) I find myself wondering if I understand all the scenarios we're trying to catch/protect against with that check (https://github.com/badges/shields/pull/6160/files#r575695319). I gather these are the three(ish) :

* flag any usage of `request`-specific options that we haven't yet properly mapped to `got`'s equivalent
* prevent the introduction of any direct `got` option usage/mixed usage of request and got options
* structure the conversion to `got` in batches, specifically here updating the service class fetch params for options

If so and I'm following correctly then I think I'm on board with that. I suppose one thing that's at least worth brief consideration is whether we want to abstract away the http client lib request options (http request, not `request` lib) from the service classes as a long term approach (not just for conversions).

I'm not proposing that we do, and think I would probably vote against, but tossing it out there anyway. I think the frequency (or lack thereof really) at which the underlying lib is changed, and the benefits of using a popular/well-known lib which many folks will know/can easily lookup outweigh trying to have our own Shields-specific set of options and translation layer

Reply:
I agree with that summary :+1: 

I think unless we've been very unlucky in backing the wrong horse here, changing what HTTP lib we are using should be something that we will not do any time soon and there isn't any benefit in building a permanent abstraction over got. This should just be here temporarily for transitional purposes.